solidity/core/contracts/AbacusConnectionManager.sol

```diff
@@ -73,6 +78,17 @@ contract AbacusConnectionManager is IAbacusConnectionManager, Ownable {
      */
     function enrollInbox(uint32 _domain, address _inbox) external onlyOwner {
         require(!isInbox(_inbox), "already inbox");
+
+        // prevent enrolling an inbox that matches any historical domain hash
+        IInbox inbox = IInbox(_inbox);
+        address vm = inbox.validatorManager();
+        bytes32 domainHash = IMultisigValidatorManager(vm).domainHash();
+        require(
+            !domainHashesEverUsed.contains(domainHash),
```

Comment:
Do we want to support enrolling, unenrolling, and re-enrolling the same inbox? As is, you wouldn't be able to re-enroll.

I think that's okay, but just flagging